src/shared/mkfs-util.c

```diff
@@ -147,15 +148,28 @@ static int do_mcopy(const char *node, const char *root) {
         /* mcopy copies the top level directory instead of everything in it so we have to pass all
          * the subdirectories to mcopy instead to end up with the correct directory structure. */
 
-        rootdir = opendir(root);
-        if (!rootdir)
-                return log_error_errno(errno, "Failed to open directory '%s'", root);
+        rfd = open(root, O_RDONLY|O_DIRECTORY|O_CLOEXEC|O_NOFOLLOW);
```

Comment:
O_NOFOLLOW should not be necesary. seems unnecessary to block symlinks for the root dir itself, but i also don't really care ;-)